### DISCUSSION TITLE
Rider not detecting tests

### DISCUSSION THREAD

**Discussion by slinkynexus** [thomhurst/TUnit#806] [Q&A]
After a day of attempts, I have not been able to get Rider integration working for TUnit. I have followed the instructions on the documentation page, as well as many attempts with a tests.runsettings in the project. I can run dotnet test from the terminal inside of Rider and that is successful, just nothing I do can make it show up in the Rider test window.

I am using:
Linux 6.10.11
Rider 2024.2.4 (Build #RD-242.21829.210, built on September 6, 2024)
dotnet --version 8.0.300

How can I get this to work?

**Comment by thomhurst** [comment]
There's a current issue with Rider. It should be fixed in their next build.

https://github.com/thomhurst/TUnit/issues/576

**Comment by JHMoneIQ** [comment]
It isn't fixed in 2024.3.3

**Comment by codyskidmore** [comment]
Head's up folks. In my install, Rider detects the tests, but dotCover does not work. I haven't figured out an alternative. I'm using JetBrains Rider 2024.3.3, .NET 9.0.100, & TUnit 0.6.117.
![image](https://github.com/user-attachments/assets/0001ad4a-0310-45ad-9323-02696ee873ff)
![image](https://github.com/user-attachments/assets/8c0c9018-d4b1-433a-aaf6-f92cb1c2837a)

**Reply by nicholass-alcidion** [reply]
I am also able to see and run tests but not coverage - any one have ideas on where to next for getting coverage? Is it nesscaary to raise a issue with Jetbrains?

**Reply by thomhurst** [reply]
There's already an issue I believe. The standalone dotcover works I've heard from another user

**Reply by nicholass-alcidion** [reply]
Found the ticket, https://youtrack.jetbrains.com/issue/DCVR-12871/Not-receiving-coverage-from-TUnit-Testing-Platform-tests-in-dotCover for anyone else looking

**Comment by codyskidmore** [comment]
@thomhurst, @nicholass-alcidion, 

Yes, the standalone version of dotCover does work correctly.

![image](https://github.com/user-attachments/assets/d0810533-b280-4c9d-a7b1-8a8d936902b4)

**Comment by mmo80** [comment]
This is still not working for me... no TUnit test are shown in Rider Tests Explorer.

OS: Fedora 43

JetBrains Rider 2025.3.0.3
TUnit 1.4.0
.NET 9.0

No issues running it from Rider terminal
<img width="1502" height="625" alt="image" src="https://github.com/user-attachments/assets/3197da74-4910-4e3a-a583-0dfbbaf69946" />

**Reply by thomhurst** [reply]
@mmo80 Make sure these two are selected in the settings: 

<img width="1052" height="278" alt="image" src="https://github.com/user-attachments/assets/5baa661b-9be5-4ad3-861f-feb670c9ef0d" />

**Reply by thomhurst** [reply]
Actually maybe not the bottom one

**Comment by mmo80** [comment]
I did as you said, selected both
<img width="602" height="155" alt="image" src="https://github.com/user-attachments/assets/9a76b69f-e296-44c4-a1b3-02f5b2604404" />

then refreshed... nothing... restarted Rider.. nothing.
<img width="447" height="62" alt="image" src="https://github.com/user-attachments/assets/9e9802f9-a368-490c-84e2-dc453bca00db" />

Later i remembered that i tinkered alot to try to get Rider to discover the tests and just to test i unselected
<img width="577" height="102" alt="image" src="https://github.com/user-attachments/assets/46541f70-7093-42bd-ae23-94b2a40a1a64" />

and then refreshed.... and BAM! it worked! So thank you @thomhurst !

**Reply by thomhurst** [reply]
You might be able to re-enable that (if you still use xUnit NUnit etc in other projects) and just uncheck the ignore projects option like:

<img width="997" height="230" alt="image" src="https://github.com/user-attachments/assets/23bf4930-9539-447f-9e9e-eedd874c2ebb" />

But glad it's working now 😄

**Reply by MihailsKuzmins** [reply]
Yeah, Rider can run tests for the entire project, but there is no Play button next to each test method.

**Reply by thomhurst** [reply]
@MihailsKuzmins please raise an issue to Jetbrains for that

**Reply by MihailsKuzmins** [reply]
yeah, I didn't mean it as the issue of the library, just observation, sorry for misunderstanding

**Reply by MihailsKuzmins** [reply]
UPDATE: actually, after rebooting my PC it started to work (restarted Rider 3 times before it - nothing has changed), so only after the PC restart the green play button has appeared. So it seems like everything is working 🥳.
